### ISSUE TITLE
Platform

### ISSUE THREAD

**Issue by KittyGiraudel** [at-import/canisass#2]
Jekyll?

**Comment by Snugug** [comment]
I'm thinking a node setup. I really hate working in Ruby

**Comment by KittyGiraudel** [comment]
Well, Jekyll doesn't real make you work with Ruby. :P

**Comment by Snugug** [comment]
Let me rephrase, I'm envisioning an automated test and build process, not us manually updating markdown files

**Comment by Snugug** [comment]
Here are my thoughts as to how I'd like to see this work:
- Tests are grouped into suites based on the functionality they are testing (`@each` with maps, for instance, would be grouped under `@each`, not maps as it's testing how `@each` works with maps)
- Do not use a testing framework written in Sass, have an input Sass file and and output CSS file. We want to test Sass, not a testing framework. For instance, you'd have `each`, `each/controls`, `each/scss`, `each/sass` to cover the different syntaxes.
- Each file name should be a descriptive slugified explanation of the single piece of functionality its testing, for instance `each-with-key-value`. The Sass files and CSS files will both be named the same.
- Running the tests will first compile the `scss` versions, compare the compiled output to the controls, and if there is a difference, generate a `.diff` file in the form of `test-name__compiler--scss` with `compiler` being the name of the compiler being run (`ruby-3-4-0`, `node-sass-1-0-3`, etc…). Then the same will be repeated for the `sass` files, swapping `scss` for `sass` in the `.diff` filename. A JSON file would be generated with the results to be able to be fed into the frontend.
- We would be able to group each test suite and show users whether a feature is fully supported or partially supported in each implementation, including `diffs` of what they can expect for failing tests.

Considering the many runtime platforms we're going to need to support, I think we're probably going to want to do a good portion of this inside a VM so we have a controlled dev environment.

I'm open to suggestions as to how to test legacy versions of compilers.

**Comment by KittyGiraudel** [comment]
> Let me rephrase, I'm envisioning an automated test and build process, not us manually updating markdown files

I think it was obvious. :)

Everything else you said sounds wise to me. Can't see any specific flaw in your system.

**Comment by KittyGiraudel** [comment]
What's the status on this?

**Comment by Snugug** [comment]
I haven’t had time to work on this yet. Probably over the holidays.
